src/state.rs

```diff
@@ -125,6 +122,22 @@ impl From<EntryPointOffset> for String {
     }
 }
 
+pub fn num_or_string<'de, D: Deserializer<'de>>(deserializer: D) -> Result<usize, D::Error> {
+    let usize_value = match Value::deserialize(deserializer)? {
+        Value::Number(number) => {
+            number.as_u64().ok_or(DeserializationError::custom("Cannot cast number to usize."))?
+                as usize
+        }
+        Value::String(s) => hex_str_try_into_usize(&s).map_err(DeserializationError::custom)?,
+        _ => return Err(DeserializationError::custom("Cannot cast value into usize.")),
+    };
+    Ok(usize_value)
+}
+
```

Comment:
Can you move the implementation of `hex_str_try_into_usize` into TryFrom<String> and use try_from inside num_or_string?